registry.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Jython Registry
+---
+
+# Jython Registry
+
+Because there is no good platform independent equivalent of the Windows Registry (or Unix envrionment variables) Java has it's own environment variable namespace. Jython aquires it's namespace from the following sources (later sources override defaults found in earlier places).
+
+ - The Java system properties: typically passed in on the command line as options to the java interpreter.
+ - The Jython "registry" file, which contains prop=value pairs. See below for the algorithm Jython uses to find the registry file.
+ - The user's personal registry file, which contains similarly formated prop/value pairs. The user's registry file is at `"user.home"+"/.jython"`
+ - Jython properties: Specified on the command line as options to the jython class. See the -D option to the interpreter.
+
+## Registry Properties
+
+The following properties are recognized by Jython. More detailed documentation may be found in comments in the registry file provided with the Jython distribution. 
+
+**python.path**
+Equivalent to CPython's PYTHONPATH environment variable
+
+**python.cachedir**
+The directory to use for caches - currently just package information. This directory must be writable by the user. If the directory is an absolute path, it is used as given, otherwise it is interpreted as relative to sys.prefix.
+
+**python.verbose**
+Sets the verbosity level for varying degrees of informative messages. Valid values in order of increasing verbosity are "error", "warning", "message", "comment", "debug"
+
+**python.security.respectJavaAccessibility**
+Normally, Jython can only provide access to public members of classes. However if this property is set to false and you are using Java 1.2 then Jython can access non-public fields, methods, and constructors.
+
```

Comment:
``Java 1.2`` seems to be an outdated hint. This should at least be ``Java < 9``.
In Java 9 non-public access is still be possible using ``--permit-illegal-access``. I don't know about Java 10+ support. Maybe this property will need deprecation.
See discussions at https://bugs.jython.org/issue2656 and https://bugs.jython.org/issue2582.
I suggest to include these notes in some form here.

Reply:
I've added a longer note. 

We have a golden source problem with the registry documentation. This page was a lightly edited import from the old website, and it says "see the config file for more details". That is fine if it's consistent. It is close to point of use. That's Option 1, and would imply a jython source patch, which I guess I can do. 

An alternative I'd personally prefer is less documentation in the registry comments (as it's the only such documentation of that type), and moving them to a new constants class, something like RegistryStandardKeys, with detailed javadoc for each property. That is slightly harder to forget when adding a new one. A property could also be explicitly `@deprecated` using the usual Java convention. We could then point all other references, including this page, to the Javadoc. This could be Option 2.

There could be other options, I guess I just think consolidated golden sources are most important.

Reply:
I am ("naively", without thinking it over and over) +1 on moving this to Javadoc. IMO it becomes much more visible that way. OTOH having some doc in the registry also feels natural. Having it in redundant form feels wrong, clearly speaking: We should absolutely avoid additional redundancy. Dealing with ``__doc__`` vs ``/**javadoc*/`` is already enough pain.
Difficult decision. @jeff5  what do you think?
Maybe some script that automates updating doc from registry to everywhere else? But the doc is so rarely changed that this wouldn't pay off.

Reply:
I surmise the registry is the way it is because it invites you to edit it (or a copy), a bit like the `conf.py` that `sphinx-quickstart` gives you and that is full of comments. But documenting them via Java(doc) is a good idea. I assume we mean to define `String`constants as they are now, just together in a class. I favour something brief like `org.python.RegistryKey`. The default value is still in the installed file registry. Or do we mean them to be alongside the name and documentation somehow?

It should still be ok for the more obscure settings to be in the code they affect, I think. `cpython2` may be one.

I note that Javadoc gives special treatment to public constants, so it would appear [like this](https://static.javadoc.io/org.python/jython-standalone/2.7.1/constant-values.html#org.python.core.PySystemState.PYTHON_CONSOLE_ENCODING), but with actual documentation.

Reply:
Ok, thanks all. Will take this guidance into a patch for `org.python.RegistryKey` which will just centralize string constants for keys and javadoc for their meanings. First version unlikely to have anything like generating `registry`, just for incremental time reasons.